### PR TITLE
Add `ruff check`, `ruff format --check`, and `pytest` Github actions

### LINKED ISSUE #152: Add linting and formatting via `ruff`
Use [ruff](https://github.com/astral-sh/ruff) for linting and formatting.

### PR DESCRIPTION
**Pull request by wiseodd**
Closes #152 

Self-explanatory!

### CODE REVIEW

**Comment by wiseodd** [comment]
TODO for self: Add "Contributing" section in the README.

File: pyproject.toml [hunk 1]
```diff
@@ -5,5 +5,9 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[tool.ruff.lint]
```
Comment (runame) [inline pyproject.toml:8]: I think it makes sense to include `isort` and `pylint` as well (plus the minor adjustments shown here):
```
[tool.ruff.lint]
# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
# McCabe complexity (`C901`) by default.
# Adds isort ("I") and pylint ("PL") to defaults.
select = ["E4", "E7", "E9", "F", "I", "PL"]
# Ignore pylint's overwritten loop variable by assignment error
ignore = ["PLW2901"]

[tool.ruff.lint.pylint]
max-args = 10
```

**Review by runame — COMMENTED** [review]
Cool! See my suggestion for the linter settings.

File: pyproject.toml (hunk 1 above)
Comment (wiseodd) [reply pyproject.toml:5]: @runame please check if you're OK with this:

1. Only enable "PLE" and "PLW" ("PLR" is disabled)
2. Quote-style: "double" --- just follow `black` as is

File: laplace/utils/feature_extractor.py [hunk 2]
```diff
@@ -105,9 +105,13 @@ def forward_with_features(
             n_intermediate_dims = len(features.shape) - 2
 
             if self.feature_reduction == FeatureReduction.PICK_FIRST:
```
Comment (wiseodd) [inline laplace/utils/feature_extractor.py:107]: @runame please review this change which will fix #195

**Review by runame — APPROVED** [review]
LGTM

**Pull request merged**